app-backend/common/src/main/scala/cache/package.scala

```diff
@@ -0,0 +1,27 @@
+package com.azavea.rf.common
+
+import net.spy.memcached._
+
+import java.util.concurrent.{Executors, TimeUnit}
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.util._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+package object cache {
+  implicit class MemcachedClientMethods(client: MemcachedClient) {
+    def getOrSet[CachedType](cacheKey: String, expensiveOperation: String => Future[CachedType], ttl: Duration)(implicit ec: ExecutionContext): Future[CachedType] = {
```

Comment:
`getOrElseUpdate` is a method from scala collections API, might as well re-use it here.